IPython/terminal/interactiveshell.py

```diff
@@ -1,810 +1,18 @@
-            except:
-                self.showtraceback()
-            else:
-                try:
-                    f = open(err.filename)
-                    try:
-                        # This should be inside a display_trap block and I
-                        # think it is.
-                        sys.displayhook(f.read())
-                    finally:
-                        f.close()
-                except:
-                    self.showtraceback()
-
-    def _should_recompile(self,e):
-        """Utility routine for edit_syntax_error"""
-
-        if e.filename in ('<ipython console>','<input>','<string>',
-                          '<console>','<BackgroundJob compilation>',
-                          None):
-
-            return False
-        try:
-            if (self.autoedit_syntax and
-                not self.ask_yes_no('Return to editor to correct syntax error? '
-                              '[Y/n] ','y')):
-                return False
-        except EOFError:
-            return False
-
-        def int0(x):
-            try:
-                return int(x)
-            except TypeError:
-                return 0
-        # always pass integer line and offset values to editor hook
-        try:
-            self.hooks.fix_error_editor(e.filename,
-                int0(e.lineno),int0(e.offset),e.msg)
-        except TryNext:
-            warn('Could not open editor')
-            return False
-        return True
-
-    #-------------------------------------------------------------------------
-    # Things related to exiting
-    #-------------------------------------------------------------------------
-
-    def ask_exit(self):
-        """ Ask the shell to exit. Can be overiden and used as a callback. """
-        self.exit_now = True
-
-    def exit(self):
-        """Handle interactive exit.
-
-        This method calls the ask_exit callback."""
-        if self.confirm_exit:
-            if self.ask_yes_no('Do you really want to exit ([y]/n)?','y','n'):
-                self.ask_exit()
-        else:
-            self.ask_exit()
-
-    #-------------------------------------------------------------------------
-    # Things related to magics
-    #-------------------------------------------------------------------------
-
-    def init_magics(self):
-        super(TerminalInteractiveShell, self).init_magics()
-        self.register_magics(TerminalMagics)
-
-    def showindentationerror(self):
-        super(TerminalInteractiveShell, self).showindentationerror()
-        if not self.using_paste_magics:
-            print("If you want to paste code into IPython, try the "
-                "%paste and %cpaste magic functions.")
-
-
-InteractiveShellABC.register(TerminalInteractiveShell)
+class TerminalInteractiveShell(PromptToolkitShell):
+    def __init__(self, *args, **kwargs):
+        warn("This is a deprecated alias for IPython.terminal.ptshell.TerminalInteractiveShell. "
```

Comment:
Want to also warn on **import** ? Can be done in separate PR + since when it is deprecated.